xdsl/dialects/onnx.py

```diff
@@ -100,12 +100,43 @@ class Div(ElementwiseBinOpBase):
     name = "onnx.Div"
 
 
+
+
+@irdl_op_definition
+class Relu(IRDLOperation):
+    name = "onnx.Relu"
+    """
+    Relu takes one input data (Tensor) and produces one output data (Tensor) where the rectified linear function,
+     y = max(0, x), is applied to the tensor elementwise.
+    """
+    T = Annotated[AnyFloat | IntegerType, ConstraintVar("T")]
+    operand = operand_def(TensorType[T])
+    res = result_def(TensorType[T])
+    assembly_format = "`(` $operand`)` attr-dict `:` `(` type($operand) `)` `->` type($res)"
+
+    def __init__(self, operand: SSAValue):
+        super().__init__(
+            operands=[operand],
+            result_types=[operand.type],
+        )
+
+    def verify_(self) -> None:
+        if (
+                not isinstance(self.operand.type, TensorType)
+                or not isinstance(self.res.type, TensorType)
+                or self.operand.type != self.res.type
+        ):
+            assert (
+                False
+            ), "onnx elementwise operation operand and result must be of type TensorType"
```

Comment:
pyright is a bit pedantic here, because the types are generic, and the generic parameter is unknown. We can actually avoid the whole verification thing with a ConstraintVar

Reply:
Oh no actually we can't do it with a constraint var, I'd recommend something like this:

```suggestion
        if (
                not isinstance(operand_type := self.operand.type, TensorType)
                or not isinstance(res_type := self.res.type, TensorType)
        ):
            assert (
                False
            ), "onnx elementwise operation operand and result must be of type TensorType"
            operand_type = cast(TensorType[Attribute], operand_type)
            res_type = cast(TensorType[Attribute], res_type)
            
            if operand_type != res_type:
                raise VerificationException("Mismatch between operand type and res type")
```